src/sage.js

```diff
@@ -131,33 +136,28 @@ class Sage {
           var transaction = {
             connection: connection,
             commit: function() {
-              sage.commit(this.connection).then(function() {
-                resolve();
-              });
+              console.log('commiting thru transaction');
+              sage.commit(this.connection).then(resolve).catch(reject);
             },
             rollback: function(transaction) {
-              sage.releaseConnection(this.connection).then(function() {
-                resolve();
-              });
+              sage.releaseConnection(this.connection).then(resolve).catch(reject);
             }
           }
           fn(transaction);
         });
       });
     } else {
-      return new Promise(function(resolve, reject) {
-        self.getConnection().then(function(connection) {
-          var transaction = {
-            connection: connection,
-            commit: function() {
-              return sage.commit(this.connection);
-            },
-            rollback: function(transaction) {
-              return sage.releaseConnection(this.connection);
-            }
+      return self.getConnection().then(function(connection) {
+        var transaction = {
+          connection: connection,
+          commit: function() {
+            return sage.commit(this.connection);
+          },
+          rollback: function(transaction) {
+            return sage.releaseConnection(this.connection);
           }
-          resolve(transaction);
-        });
+        }
+        return Promise.resolve(transaction);
```

Comment:
I think you just want `return transaction` here, you don't have to return a resolved promise from a `then`